src/Agent.Plugins/GitCliManager.cs

```diff
@@ -175,7 +175,7 @@ public async Task<int> GitFetch(AgentTaskPluginExecutionContext context, string
             string progress = reducedOutput ? string.Empty : "--progress";
 
             // insert prune-tags if knob is false to sync tags with the remote
-            string pruneTags = AgentKnobs.DisableFetchPruneTags.GetValue(context).AsBoolean() ? string.Empty : "--prune-tags";
+            string pruneTags = !EnsureGitVersion(new Version(2, 17), throwOnNotMatch: false) ? string.Empty : AgentKnobs.DisableFetchPruneTags.GetValue(context).AsBoolean() ? string.Empty : "--prune-tags";
```

Comment:
Can we make it a little more readable?

```suggestion
            string pruneTags = string.Empty;
            if (EnsureGitVersion(new Version(2, 17), throwOnNotMatch: false) && !AgentKnobs.DisableFetchPruneTags.GetValue(context).AsBoolean())
            {
                pruneTags = "--prune-tags";
            }
```